contentcuration/contentcuration/frontend/settings/vuex/index.js

```diff
@@ -78,14 +82,37 @@ export default {
       return client.post(window.Urls.delete_user_account(), { email });
     },
 
-    // Fetch fields that take longer to calculate
-    fetchDeferredUserData(context) {
+    // Fetch the user API token
+    fetchDeferredUserApiToken(context) {
       if (context.getters.storageUseByKind && context.state.currentUser.api_token) {
```

Comment:
```
      if (context.getters.storageUseByKind && context.state.currentUser.api_token) {
```
I am worried about using this same code twice, there must be a better way!

If I just check `context.state.currentUser.api_token` I get thjs error: 
![Screenshot from 2023-03-12 22-24-25](https://user-images.githubusercontent.com/53405133/224559880-70d9e2b5-b71f-4a72-b923-22481e3e4d5c.png)

Reply:
@akash5100 In `fetchDeferredUserApiToken(context)` you just need to check for API_TOKEN by using `context.state.currentUser.api_token` and in `fetchDeferredUserStorageByKind(context)` you just need to check for `context.getters.storageUseByKind`.

Reply:
`if (context.state.currentUser.api_token)` throws the above TypeError.

Reply:
@akash5100 we can use `try..catch..` here to catch that error? 

@bjester thoughts sir?

Reply:
Can this be a possible fix, here we first check if `context.state` has a `currentUser`, if yes then we check for `api_token`?
```
if (context.state.currentUser && context.state.currentUser.api_token) {
  // 
}
```

Reply:
First, the intent of the if statement is to avoid repeat backend requests when the data is already loaded into Vuex. Second, since you're splitting the two API requests, the if statement also needs split appropriately. Lastly, I believe the error is exposing that this code was actually incorrect to begin with since `currentUser` should always be an object. These are inside a namespaced vuex module, so in order to access those other vuex bits, it needs to use `rootGetters` and `rootState`:

```
if (context.rootGetters.storageUseByKind)
```
```
if (context.rootState.session.currentUser.api_token)
```